### PR TITLE
Update P + S WDL to get around input limits [VS-970]

### PR DESCRIPTION
**Pull request by mcovarr**
Successful run [here](https://app.terra.bio/#workspaces/allofus-drc-wgs-dev/NHGRI_AnVIL_3K_AoU_Development/job_history/b5f0bfac-d0cb-490c-b521-fd809b4bdeb3) (requires PMI ops). Takes a FOFN input of full cloud paths to output VCFs for control samples instead of a literal array.

### CODE REVIEW

File: scripts/variantstore/wdl/GvsCalculatePrecisionAndSensitivity.wdl [hunk 1]
```diff
@@ -120,47 +129,56 @@ workflow GvsCalculatePrecisionAndSensitivity {
...
-        echo "Skipping ~{input_vcf} as it is not on $i."
+        touch output/${output_vcf_name}
+        echo "Skipping ${input_vcf_path} as it is not on $i."
       fi
     done
   >>>
 
   runtime {
-    docker: "gcr.io/gcp-runtimes/ubuntu_16_0_4:latest"
+    docker: "us.gcr.io/broad-dsde-methods/variantstore:2023-07-24-alpine-bd6b9d62e"
```
Comment (rsasch) [inline scripts/variantstore/wdl/GvsCalculatePrecisionAndSensitivity.wdl:175]: Do we need to use our docker?  Wouldn't gcr.io/google.com/cloudsdktool/cloud-sdk:426.0.0-alpine be smaller and also work?

**Review by rsasch — APPROVED** [review]
One small question, otherwise 👍🏻

File: scripts/variantstore/wdl/GvsCalculatePrecisionAndSensitivity.wdl [hunk 2]
```diff
@@ -451,16 +469,17 @@ task CollateReports {
...
+
+task CountInputVcfs {
   input {
-    String message
+    File input_vcf_fofn
   }
   command <<<
-    >&2 echo "Error: ~{message}"
-    exit 1
+    wc -l < ~{input_vcf_fofn} > num_vcfs.txt
```
Comment (RoriCremer) [inline scripts/variantstore/wdl/GvsCalculatePrecisionAndSensitivity.wdl:478]: what does the < do here?

**Review by RoriCremer — APPROVED** [review]

File: scripts/variantstore/tieout/AoU_PRECISION_SENSITIVITY.md [hunk 3]
```diff
@@ -7,16 +7,19 @@ Complete (at least) through the training step (`GvsCreateFilterSet`) of the [GVS
 
 The wdl takes several inputs:
 
-**input_vcfs** - A collection of VCFS - generated by `GvsExtractCallSet`. These need not be subsetted down to the chromosome.
+**input_vcf_fofn** - A FOFN (file of file names) of output VCFs for control samples generated by `GvsExtractCallSet`. These need not be subsetted down to the chromosome.
```
Comment (RoriCremer) [inline scripts/variantstore/tieout/AoU_PRECISION_SENSITIVITY.md:10]: doesn't our extract automatically split by chr?

File: scripts/variantstore/wdl/GvsCalculatePrecisionAndSensitivity.wdl (hunk 1 above)
Comment (mcovarr) [reply scripts/variantstore/wdl/GvsCalculatePrecisionAndSensitivity.wdl:175]: Oh probably so; I needed to step up from `ubuntu` to get `gcloud`, but I don't think I need all the other stuff in our image.

File: scripts/variantstore/wdl/GvsCalculatePrecisionAndSensitivity.wdl (hunk 2 above)
Comment (mcovarr) [reply scripts/variantstore/wdl/GvsCalculatePrecisionAndSensitivity.wdl:478]: `<` redirects the file to be the standard input of `wc`. When `wc` reads from standard input it emits a bare(ish) number delta some whitespace that `read_int` ignores:

```
% wc -l < foo.txt
       3
%
```

But with the file as an argument, `wc` "helpfully" prints the name of the file which `read_int` does not like at all:

```
% wc -l foo.txt  
       3 foo.txt
%
```

File: scripts/variantstore/tieout/AoU_PRECISION_SENSITIVITY.md (hunk 3 above)
Comment (mcovarr) [reply scripts/variantstore/tieout/AoU_PRECISION_SENSITIVITY.md:10]: I think that's right. But the P+S logic doesn't seem to require that it receive VCFs containing only data for the chromosomes on which it's trying to calculate P+S.

**Review by gbggrant — APPROVED** [review]

**Pull request merged**